src/webportal/src/app/user/fabric/userView/TopBar.jsx

```diff
@@ -0,0 +1,352 @@
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import React, {useContext, useMemo, useState} from 'react';
+
+import {CommandBarButton, SearchBox, CommandBar, ContextualMenuItemType} from 'office-ui-fabric-react';
+import {findIndex} from 'lodash';
+
+import Context from './Context';
+import Filter from './Filter';
+import {toBool} from './utils';
+
+/* eslint-disable react/prop-types */
+function FilterButton({defaultRender: Button, ...props}) {
+  const {subMenuProps: {items}} = props;
+  const checkedItems = items.filter((item) => item.checked).map((item) => item.text);
+  const checkedText = checkedItems.length === 0 ? null
+    : checkedItems.length === 1 ? <strong>{checkedItems[0]}</strong>
+      : <strong>{checkedItems[0]}{` (+${checkedItems.length - 1})`}</strong>;
+  return (
+    <Button {...props}>
+      {checkedText}
+    </Button>
+  );
+}
+
+function KeywordSearchBox() {
+  const {filter, setFilter} = useContext(Context);
+  function onKeywordChange(keyword) {
+    const {admins, virtualClusters} = filter;
+    setFilter(new Filter(keyword, admins, virtualClusters));
+  }
+
+  /** @type {import('office-ui-fabric-react').IStyle} */
+  const rootStyles = {backgroundColor: 'transparent', alignSelf: 'center', width: 220};
+  return (
+    <SearchBox
+      underlined
+      placeholder="Filter by keyword"
+      styles={{root: rootStyles}}
+      value={filter.keyword}
+      onChange={onKeywordChange}
+    />
+  );
+}
+/* eslint-enable react/prop-types */
+
+function TopBar() {
+  const [active, setActive] = useState(true);
+  const {allUsers, refreshAllUsers, getSelectedUsers, filter, setFilter, addUser, importCSV, removeUsers} = useContext(Context);
+
+  const {admins, virtualClusters} = useMemo(() => {
+    const admins = Object.create(null);
+    const virtualClusters = Object.create(null);
+
+    if (allUsers !== null) {
+      allUsers.forEach(function(user) {
+        admins[String(user.admin)] = true;
+        if (user.virtualCluster) {
+          const vcs = user.virtualCluster.split(',');
+          vcs.forEach((vc) => {
+            virtualClusters[vc] = true;
+          });
+        }
+      });
+    }
+
+    return {admins, virtualClusters};
+  }, [allUsers]);
+
+  /**
+   * @type {import('office-ui-fabric-react').ICommandBarItemProps}
+   */
+  const btnAddUser = {
+    key: 'addUser',
+    name: 'Add User',
+    buttonStyles: {root: {backgroundColor: 'transparent', height: '100%'}},
```

Comment:
100% height in flexbox container may not work on an old version of chrome.

Reply:
Verified on Chrome 63 and it works.
Double checked that only the button `Clear` need this property, so remove this property from other buttons.